common/token-generation-lib/src/main/java/org/eclipse/dataspaceconnector/common/token/TokenGenerationServiceImpl.java

```diff
@@ -19,17 +19,32 @@
 import com.nimbusds.jose.JWSHeader;
 import com.nimbusds.jose.JWSSigner;
 import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
 import com.nimbusds.jwt.JWTClaimsSet;
 import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.spi.EdcException;
 import org.eclipse.dataspaceconnector.spi.iam.TokenRepresentation;
 import org.eclipse.dataspaceconnector.spi.result.Result;
 import org.jetbrains.annotations.NotNull;
 
+import java.security.PrivateKey;
+import java.security.interfaces.ECPrivateKey;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Supplier;
+
 public class TokenGenerationServiceImpl implements TokenGenerationService {
 
+    private static final String KEY_ALGO_RSA = "RSA";
+    private static final String KEY_ALGO_EC = "EC";
+
     private final JWSSigner tokenSigner;
     private final JWSAlgorithm jwsAlgorithm;
 
+    public TokenGenerationServiceImpl(Supplier<PrivateKey> privateKeySupplier) {
```

Comment:
since the supplier it's evaluated on construction, there's no specific need for it, `PrivateKey` could be passed directly